File: car_class.rb
class Car
    attr_reader :make, :model, :year, :color, :speed

    def initialize(make, model, year, color)
        @make = make
        @model = model
        @year = year
        @color = color
        @speed = 0
    end

    def accelerate(increment)
        @speed += increment
    end

    def brake(decrement)
        @speed -= decrement
    end

    def display_speed
        puts "The current speed of the #{color} #{make} #{model} is #{@speed} mph.\n"
    end
end

my_car = Car.new("Ford", "F-150", 2022, "Gray")

puts "My car is a #{my_car.year} #{my_car.color} #{my_car.make} #{my_car.model}."

my_car.accelerate(30)
my_car.display_speed

my_car.brake(10)
my_car.display_speed

File: car_class.py
# It has been a while since 2150 I can't find the original prompt.

class Car:
    def __init__(self, make, model, year, color):
        self.make = make
        self.model = model
        self.year = year
        self.color = color
        self.speed = 0
    
    def accelerate(self, increment):
        self.speed += increment
    
    def brake(self, decrement):
        self.speed -= decrement
    
    def display_speed(self):
        print(f"The current speed of the {self.color} {self.make} {self.model} is {self.speed} mph.")


# Example usage
my_car = Car("Ford", "F-150", 2022, "Gray")

print(f"My car is a {my_car.year} {my_car.color} {my_car.make} {my_car.model}.")

my_car.accelerate(30)
my_car.display_speed()

my_car.brake(10)
my_car.display_speed()
